build-system/compile/internal-version.js

```diff
@@ -16,50 +16,97 @@
+ *    - followed by two commits that were created directly on the branch, the
+ *      last of which was commited on November 23, 2021 6:38 PM PST,
+ *    - followed by twelve commits that were cherry-picked from `master`, then
+ *      its version number would be `2111240238012`.
+ *    - The latest twelve commits are cherry-picks from `master`, and the one
+ *      before them is not, so our last three digits are set to 012.
+ *    - PST is UTC-8, so the hour value changes from PST's 18 to UTC's 2, and
+ *      one day is added.
+ *
+ * The version number can be manually overridden by passing --version_override
+ * to the `gulp build`/`gulp dist` command.
+ *
+ * @return {string} AMP version number (always 13 digits long)
+ */
 function getVersion() {
   if (argv.version_override) {
     const version = String(argv.version_override);
-    // #27579: What are allowed version strings...?
     if (!/^\d{13}$/.test(version)) {
       throw new Error('--version_override only accepts a 13-digit version');
     }
     return version;
-  } else {
-    // Generate a consistent version number by using the commit* time of the
-    // latest commit on the active branch as the twelve digits. The last,
-    // thirteenth digit defaults to 0, but can be changed by setting the
-    // --custom_version_mark flag to a different value. This should rarely be
-    // used by AMP release engineers.
-    //
-    // e.g., the version number of a clean (no uncommited changes) tree that was
-    // commited on August 1, 2018 at 14:31:11 EDT would be `1808011831110`
-    // (notice that due to timezone shift, the hour value changes from EDT's 14
-    // to UTC's 18. The last digit is the default value of 0 as
-    // --custom_version_mark was not set.)
-    //
-    // *Commit time is different from author time! Commit time is the time that
-    // the PR was merged into master; author time is when the author ran the
-    // "git commit" command.
-    const lastCommitFormattedTime = gitCommitFormattedTime();
-    let lastDigit = 0;
-    if (argv.custom_version_mark) {
-      lastDigit = parseInt(argv.custom_version_mark, 10);
-      if (isNaN(lastDigit) || lastDigit < 0 || lastDigit > 9) {
-        throw new Error(
-          `--custom_version_mark is set to ${argv.custom_version_mark}, ` +
-            'expected value between 0 and 9!'
-        );
-      }
+  }
+
+  let numberOfCherryPicks = 0;
+  const commitCherriesInfo = gitCherryMaster().reverse();
+  for (const {isCherryPick} of commitCherriesInfo) {
+    if (!isCherryPick) {
+      break;
     }
-    return `${lastCommitFormattedTime}${lastDigit}`;
+    numberOfCherryPicks++;
   }
+  if (numberOfCherryPicks > 999) {
+    throw new Error(
+      `This branch has ${numberOfCherryPicks} cherry-picks, which is more than 999, the maximum allowed number of cherry-picks!`
+    );
+  }
+
+  const lastCommitFormattedTime = gitCommitFormattedTime(
+    `HEAD~${numberOfCherryPicks}`
+  ).slice(0, -2);
```

Comment:
This seems to be the only caller. Why not just remove the `%S` in the formatted time?

Reply:
https://github.com/ampproject/amphtml/pull/27848#discussion_r412455890